### PR TITLE
Put React.Fragment under a feature flag

### PR DESCRIPTION
**Pull request by clemmy**
Let's feature flag the React.Fragment export so that it's not publicly consumable until 16.2.0. At this point, Typescript 2.6.2 should also be released to support this.

### CODE REVIEW

**Comment by gaearon** [comment]
Why would it be bad to let people consume it before TS supports it?

**Comment by gaearon** [comment]
People already consume third party wrappers like `react-aux` or create their own so I'd prefer that we provide an official solution with correct behavior sooner rather than later.

**Comment by clemmy** [comment]
So we can coordinate our release with Typescript. 🙂

**Comment by gaearon** [comment]
What about Babel? Is it also coming out at the same time?

It would make sense to me that *syntax* support is coordinated (i.e. Babel and TypeScript support `<>` at the same time). I’m not sure coordination matters for the manual `React.Fragment` export though. What are the pros and cons of introducing them at the same time?

**Comment by clemmy** [comment]
You bring up a good point there at @gaearon . Ideally, syntax support should be the one that's coordinated, but Babel is a third party that we'll have to introduce into the release coordination, whereas React is something we have control over. I can bring up a discussion in the Babel slack channel to see if they would be willing to let me wrap `React.Fragment` inside a feature flag.

File: packages/react/src/React.js [hunk 1]
```diff
@@ -43,7 +44,9 @@ var React = {
   Component: ReactBaseClasses.Component,
   PureComponent: ReactBaseClasses.PureComponent,
   unstable_AsyncComponent: ReactBaseClasses.AsyncComponent,
-  Fragment: REACT_FRAGMENT_TYPE,
+  Fragment: ReactFeatureFlags.enableReactFragment
```
Comment (gaearon) [inline packages/react/src/React.js]: Maybe let's not export it at all? It would be a bit odd to see it on `React` object being undefined.

**Comment by sophiebits** [comment]
> I can bring up a discussion in the Babel slack channel to see if they would be willing to let me wrap React.Fragment inside a feature flag.

What do you mean by this? Not sure I understand why Babel would care about if we have a feature flag or not.

**Comment by clemmy** [comment]
Sorry, I mean <></>.

**Pull request merged**